core/src/main/java/apoc/trigger/TriggerNewProcedures.java

```diff
@@ -23,20 +22,23 @@
 
 
 public class TriggerNewProcedures {
+    public static final String NON_SYS_DB_ERROR = "The procedure should be executed against a system database.";
     public static final String TRIGGER_NOT_ROUTED_ERROR = "The procedure should be routed and executed against a writer database.";
     public static final String TRIGGER_BAD_TARGET_ERROR = "Triggers can only be installed on user databases.";
     public static final String DB_NOT_FOUND_ERROR = "The user database with name '%s' does not exist";
 
     @Context public GraphDatabaseAPI db;
     
-    @Context public Log log;
-    
     @Context public Transaction tx;
 
-    private void checkInSystemWriter() {
+    private void checkInSystem(boolean checkWriteable) {
         TriggerHandlerNewProcedures.checkEnabled();
         
-        if (!db.databaseName().equals(SYSTEM_DATABASE_NAME) || !Util.isWriteableInstance(db)) {
+        if (!db.databaseName().equals(SYSTEM_DATABASE_NAME)) {
+            throw new RuntimeException(NON_SYS_DB_ERROR);
+        }
+        
+        if (checkWriteable && !Util.isWriteableInstance(db)) {
             throw new RuntimeException(TRIGGER_NOT_ROUTED_ERROR);
         }
     }
```

Comment:
Don't you think it would be better if we just changed the logic of the show method? This changes the logic for the rest of the trigger methods as well, so they would output a different error message now if they are executed against the neo4j database. So in show, call `checkInSystem()` that checks we are in a system database. In the rest of the trigger databases, call `checkInSystemWriter`, that check we are in a system LEADER (i.e. leave it as it was)

Reply:
@ncordon Yes that makes sense. Changed